tests/test_helpscout_discovery.py

```diff
@@ -0,0 +1,120 @@
+                expected_replication_method = self.expected_replication_method()[stream]
+
+                # collecting actual values
+                schema_and_metadata = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+                metadata = schema_and_metadata["metadata"]
+                stream_properties = [item for item in metadata if item.get("breadcrumb") == []]
+                actual_primary_keys = set(
+                    stream_properties[0].get(
+                        "metadata", {self.PRIMARY_KEYS: []}).get(self.PRIMARY_KEYS, [])
+                )
+                actual_replication_keys = set(
+                    stream_properties[0].get(
+                        "metadata", {self.REPLICATION_KEYS: []}).get(self.REPLICATION_KEYS, [])
+                )
+                actual_replication_method = stream_properties[0].get(
+                    "metadata", {self.REPLICATION_METHOD: None}).get(self.REPLICATION_METHOD)
+                actual_automatic_fields = set(
+                    item.get("breadcrumb", ["properties", None])[1] for item in metadata
+                    if item.get("metadata").get("inclusion") == "automatic"
+                )
+
+
+
+                ##########################################################################
+                ### metadata assertions
+                ##########################################################################
+
+                # verify there is only 1 top level breadcrumb in metadata
+                self.assertTrue(len(stream_properties) == 1,
+                                msg="There is NOT only one top level breadcrumb for {}".format(stream) + \
+                                "\nstream_properties | {}".format(stream_properties))
+
+                # verify primary key(s)
+                self.assertSetEqual(expected_primary_keys, actual_primary_keys, msg = f"expected primary keys is {expected_primary_keys} but actual primary keys is {actual_primary_keys}")
+
+                # TDL - 16188 : BUG : The tap renames the replication key for stream - conversations from user_updated_at to updated_at, which needs to be corrected
+
+                # verify replication key(s)
+                self.assertEqual(expected_replication_keys, actual_replication_keys, msg = f"expected replication key is {expected_replication_keys} but actual replication key is {actual_replication_keys}")
+
+                # verify replication key is present for any stream with replication method = INCREMENTAL
+                if actual_replication_method == 'INCREMENTAL':
+                    self.assertEqual(expected_replication_keys, actual_replication_keys)
+
+                # verify the primary, replication keys are given the inclusions of automatic
+                self.assertEqual(catalog['metadata']['inclusion'],'available')
+                print(stream + ' has the automatic inclusion')
```

Comment:
This is not testing what the comment indicates, we can go over this test case together.

Reply:
https://github.com/singer-io/tap-hubspot/blob/2511ce047e4da0df0baf850b89a2b52c12921631/tests/test_hubspot_discovery.py#L117-L125